scripts/getChangelog.py

```diff
@@ -1,90 +1,587 @@
+token = os.getenv('GITHUB_API_TOKEN') # The GitHub API token
+base_ref = os.getenv('BASE') # The base reference (release code or branch); point of reference
+head_ref = os.getenv('HEAD') # The head reference (release code or branch); environment containing the changes
+needs_review_reply_label = os.getenv('NEED_REVIEW_REPLY_LABEL', "Needs: Review reply") # The name or ID of the label that indicates that we should reply to a (Google Play) review when the issue or PR that has this label iss addressed or released
+client = GraphQLClient('https://api.github.com/graphql')
+
+# Function: Handle exceptions
+def handle_exception(error_message: str, data: dict, error_introduction="Error"):
+  print(f"\n{error_introduction}: {str(error_message)}")
+  if data:
+    print("JSON data:")
+    print(json.dumps(data, indent=2))
+
+# Function: Display processing animation
+animation_state = {"text": "Loading", "stop": False}
+def display_processing_animation(state):
+  print()
+  characters = ["⣷", "⣯", "⣟", "⡿", "⢿", "⣻", "⣽", "⣾"]
+  while not state['stop']:
+    for char in characters:
+      print(f"\r{char} {state['text']}...                    ", end="", flush=True)
+      time.sleep(0.5)
+  print("\r                                       ", end="\r", flush=True)
+
+# Function: Get list of PRs for a given commit
+# Note: this might be unnecessary in case commits can only have one PR
+def get_associated_prs(commitid):
+  """
+  Called in case the number of PRs linked to a commit exceeds the maximum number of PRs that the script will fetch per commit.
+  """
+  global animation_text
+  animation_text = "Get PRs linked to commit"
+  query = '''
+  query ($cursor: String, $owner: String!, $repo: String!, $commitid: GitObjectID!) { 
+    repository(name: $repo, owner: $owner) { 
+      object(oid: $commitid) {
+        ... on Commit {
+          associatedPullRequests(first: 100, after: $cursor) {
+            pageInfo {
+              endCursor
+              hasNextPage
+            }
+            nodes {
+              number
+            }
+          }
+        }
+      }
+    }
+  }
+  '''
+  # Variable definition for GraphQL Explorer:
+  # {"commitid": "863d4c3b611df83389d82958114bfd2d1204e457", "cursor": null, "owner": "AntennaPod", "repo": "AntennaPod"}
+  has_next_pr_page = True
+  cursor = None
+  pr_numbers = []  # Create PR list
+  while has_next_pr_page:
+    variables = {"cursor": cursor, "owner": owner, "repo": repo, "commitid": commitid }
+    result = client.execute(query, variables)
+    data = json.loads(result)        
+    pr_numbers.extend([pr['number'] for pr in data['data']['repository']['commit']['associatedPullRequests']['nodes']])
+    page_info = data['data']['repository']['commit']['associatedPullRequests']['pageInfo']
+    has_next_pr_page = page_info['hasNextPage']
+    cursor = page_info['endCursor']
+    
+    time.sleep(1)  # To prevent hitting rate limits
+    return pr_numbers
+
+try: # Catch KeyboardInterrupt
```

Comment:
This `try` block here spans almost 500 lines of code. That is very long. Do you have an idea on how to structure the content a bit more than this huge block?

Edit: I think moving out the function definitions would help a lot